rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoExecutor.java

```diff
@@ -306,27 +316,106 @@ protected TableRecords queryCurrentRecords(Connection conn) throws SQLException
         return currentRecords;
     }
 
+    protected List<Field> getOrderedPkList(TableRecords image,Row row,String dbType) {
+        List<Field> pkFields = new ArrayList<>();
+        // To ensure the order of the pk, the order should based on getPrimaryKeyOnlyName.
+        List<String> pkColumnNameListByOrder = image.getTableMeta().getPrimaryKeyOnlyName();
```

Comment:
Why getPrimaryKeyOnlyName() here will be orderd?

Reply:
I still have some question here.
1. you do not sort the list, why call it 'ordered'?
2.If you just want to get the same order of field in 'undo sql' and 'undo prepare', I think the order of field would be same when you get it in different time.